### ISSUE TITLE
Replace `register_noop_background_update` with `DELETE` deltas

### ISSUE THREAD

**Issue by clokep** [matrix-org/synapse#12940]
From a thread: https://github.com/matrix-org/synapse/pull/12899#discussion_r883911053

It seems that `register_noop_background_update(...)` could be replaced with `DELETE FROM background_updates WHERE name = '....';`

This would be nicer since we don't end up with dead code, `register_noop_background_update` was added in 6b02fc80d173d3d4de81623d411a136abe1637e9 and from the [conversation in #synapse-dev](https://matrix.to/#/!vcyiEtMVHIhWXcJAfl:sw1v.org/$8V9zsGr_3MD9OyUHSz5LWd4qBtcAVj508JGunH_ID4E?via=sw1v.org&via=matrix.org&via=element.io) it seems like the `DELETE` approach was not considered.

**Comment by erikjohnston** [comment]
AHA, I think I remember why we didn't do this as a delta: there have been times where we've told people to manually insert data into `background_updates` to re-run those updates for whatever reason. If they do that after the delta to delete the stale background update has run then we'll get an entry stuck in that table forever. 

Not sure how much we care about that though.

**Comment by richvdh** [comment]
> AHA, I think I remember why we didn't do this as a delta: there have been times where we've told people to manually insert data into `background_updates` to re-run those updates for whatever reason. If they do that after the delta to delete the stale background update has run then we'll get an entry stuck in that table forever.

Have we ever done this for anything other than `populate_stats_process_rooms` and `populate_stats_process_users` (which we're not getting rid of)?

**Comment by erikjohnston** [comment]
> > AHA, I think I remember why we didn't do this as a delta: there have been times where we've told people to manually insert data into `background_updates` to re-run those updates for whatever reason. If they do that after the delta to delete the stale background update has run then we'll get an entry stuck in that table forever.
> 
> Have we ever done this for anything other than `populate_stats_process_rooms` and `populate_stats_process_users` (which we're not getting rid of)?

Not sure, probably not? I think we should probably still do this change, but we should remember that after this change its no longer safe to just blindly tell people to insert stuff back into `background_updates`.

**Comment by clokep** [comment]
The only use I found that might refer to that while putting together #12954 was:

https://github.com/matrix-org/synapse/blob/3c8f1290b8ccf52462baa6b3e78d1caae2906395/synapse/storage/databases/main/stats.py#L123-L127

**Comment by clokep** [comment]
We also now have an admin API for starting background jobs that we expect folks to have to run manually sometimes: https://matrix-org.github.io/synapse/develop/usage/administration/admin_api/background_updates.html#run

The `job_name` there doesn't actually get inserted, it gets mapped to the true job names:

https://github.com/matrix-org/synapse/blob/e24ff8ebe3d4119d377355402245947f7de61c00/synapse/rest/admin/background_updates.py#L125-L147

I think this gives us a little bit of protection too?

**Comment by clokep** [comment]
@DMRobertson thought that we might ask people to insert some entries into this for the user directory, I believe we used to but now have the above admin commands.

I did find some references in our upgrade notes though:

https://github.com/matrix-org/synapse/blob/master/docs/upgrade.md#incorrect-database-migration-in-old-synapse-versions

These ask to insert the `current_state_events_membership` and `populate_stats_process_rooms` in the background update table. Neither of these are no-op right now.

I think this is safe to do, so I'm going to merge it. 👍
